internal/temporalite/lite_server.go

```diff
@@ -0,0 +1,395 @@
+package temporalite
+
+// TODO(jlegrone): Refactor this package into one or more temporal.ServerOption types.
+
+import (
+	"context"
+	"fmt"
+	"math/rand"
+	"os"
+	"path/filepath"
+	"sort"
+	"strings"
+	"time"
+
+	"go.temporal.io/sdk/client"
+
+	"go.temporal.io/server/common/authorization"
+	"go.temporal.io/server/common/cluster"
+	"go.temporal.io/server/common/config"
+	"go.temporal.io/server/common/dynamicconfig"
+	"go.temporal.io/server/common/log"
+	"go.temporal.io/server/common/metrics"
+	sqliteplugin "go.temporal.io/server/common/persistence/sql/sqlplugin/sqlite"
+	"go.temporal.io/server/schema/sqlite"
+	"go.temporal.io/server/temporal"
+)
+
+const localBroadcastAddress = "127.0.0.1"
+
+// LiteServerConfig encodes options for LiteServer instances.
+type LiteServerConfig struct {
+	// When true, Ephemeral disables file persistence and uses the in-memory storage driver.
+	// State will be reset on each process restart.
+	Ephemeral bool
+	// DatabaseFilePath persists state to the file at the specified path. If the db file does
+	// not already exist, it is created and schema migrations are automatically run.
+	//
+	// This is required if Ephemeral is false.
+	DatabaseFilePath string
+	// Address on which frontend service should listen.
+	FrontendIP string
+	// Port on which frontend service should listen.
+	FrontendPort int
+	// WithMetricsPort sets the listening port for the default Prometheus metrics handler.
+	//
+	// When unspecified, the port will be system-chosen.
+	//
+	// This field is ignored when the WithCustomMetricsHandler server option is enabled.
+	MetricsPort int
+	// Namespaces specified here will be automatically registered on Temporal start.
+	Namespaces []string
+	// SQLitePragmas specified here will be applied as pragma statements to SQLite on Temporal start.
+	SQLitePragmas map[string]string
+	// Logger overrides the default logger.
+	Logger log.Logger
+	// BaseConfig sets the default Temporal server configuration.
+	//
+	// Storage and client configuration will always be overridden, however base config can be
+	// used to enable settings like TLS or authentication.
+	//
+	// Note that ServerOption arguments can also be passed to the NewLiteServer function.
+	// Always prefer setting BaseConfig over using WithConfig however, as WithConfig overrides
+	// all LiteServer specific settings.
+	BaseConfig *config.Config
+	// DynamicConfig sets dynamic config values used by the server.
+	DynamicConfig dynamicconfig.StaticClient
+}
+
+func (cfg *LiteServerConfig) apply(serverConfig *config.Config, provider *portProvider) {
+	sqliteConfig := config.SQL{
+		PluginName:        sqliteplugin.PluginName,
+		ConnectAttributes: make(map[string]string),
+		DatabaseName:      cfg.DatabaseFilePath,
+	}
+	if cfg.Ephemeral {
+		sqliteConfig.ConnectAttributes["mode"] = "memory"
+		sqliteConfig.ConnectAttributes["cache"] = "shared"
+		sqliteConfig.DatabaseName = fmt.Sprintf("%d", rand.Intn(9999999))
```

Comment:
I think the idea here was to ensure no overlap between temporalite instances running in different goroutines, since we're running with sqlite's [shared cache](https://www.sqlite.org/sharedcache.html) mode. I'm not actually certain if this is still necessary, so added a TODO comment. I'd rather not remove this at the moment since it would require more testing to be sure that it's safe.